app/models/volunteer.rb

```diff
@@ -41,11 +46,11 @@ class Volunteer < ActiveRecord::Base
 
   # more trustworthy and self.assigned? attribute?
   def unassigned?
-    self.assignments.length == 0
+    assignments.empty?
   end
 
   def needs_training?
-    not self.logs.collect{ |l| l.complete }.any?
+    logs.none?(&:complete)
```

Comment:
> Does Rails simplify this to `.where(complete: false)`? If not, that's one query instead of N

I don't believe Rails simplifies it. It should only generate one SQL query, but it could be made more efficient by doing a `.count`. I'll update it 👍 

> there are a couple other places I see "AND NOT complete", so this could even be a scope on `Log` :)

I think @Fosome is adding this scope soon.